localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -128,41 +130,48 @@ def __init__(self, version_manager: LambdaVersionManager, event_queue_url: str):
         )
 
     def run(self, *args, **kwargs):
-        try:
-            sqs_client = get_sqs_client(
-                self.version_manager.function_version, client_config=CLIENT_CONFIG
-            )
-            function_timeout = self.version_manager.function_version.config.timeout
-            while not self._shutdown_event.is_set():
-                # TODO: Fix proper shutdown causing EndpointConnectionError
-                # https://app.circleci.com/pipelines/github/localstack/localstack/17428/workflows/391fc320-0cec-4dd1-9e3b-d7511de61d12/jobs/132663/parallel-runs/2
-                # Test case (happens not every time!):
-                # tests.aws.services.cloudformation.resources.test_legacy.TestCloudFormation.test_updating_stack_with_iam_role
-                messages = sqs_client.receive_message(
+        sqs_client = get_sqs_client(
+            self.version_manager.function_version, client_config=CLIENT_CONFIG
+        )
+        function_timeout = self.version_manager.function_version.config.timeout
+        while not self._shutdown_event.is_set():
+            try:
+                response = sqs_client.receive_message(
                     QueueUrl=self.event_queue_url,
+                    # TODO: consider replacing with short polling instead of long polling to prevent keeping connections open
+                    # however, we had some serious performance issues when tried out, so those have to be investigated first
                     WaitTimeSeconds=2,
```

Comment:
Leaving the long polling here instead of switching to short polling is the main fix for the whole of the changes.

Reply:
yeah, that's the main fix of the fix :)
For unknown reasons, short-polling caused accumulated slowdowns of 20-35min across our entire test suite 🤷